ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -629,8 +634,10 @@ private void processReplicaMetadataResponse(Set<StoreKey> missingRemoteStoreKeys
     try {
       List<StoreKey> storeKeysToConvert = new ArrayList<>();
       for (ReplicaMetadataResponseInfo replicaMetadataResponseInfo : response.getReplicaMetadataResponseInfoList()) {
-        for (MessageInfo messageInfo : replicaMetadataResponseInfo.getMessageInfoList()) {
-          storeKeysToConvert.add(messageInfo.getStoreKey());
+        if (replicaMetadataResponseInfo.getError() == ServerErrorCode.No_Error) {
+          for (MessageInfo messageInfo : replicaMetadataResponseInfo.getMessageInfoList()) {
```

Comment:
Can `ReplicaMetadataResponseInfo.getMessageInfoList()` be null even when local `getError()` is `No_Error`? (what happens if there is nothing to replicate for this replica - is the list going to be empty or null?). If it can be null, then this can throw NPE.